tests/test_contracts.py

```diff
@@ -46,14 +54,30 @@ def scaffolder(block_explorer):
 @responses.activate
 def test_scaffolder_generate(scaffolder):
     """Test the scaffolder."""
+    test_abi = {"abi": "some_abi"}
+
+    # Test mainnet scaffolding
     responses.add(
         responses.GET,
-        f"{BLOCK_EXPLORER_URL}/api?module=contract&action=getabi&address={KNOWN_ADDRESS}",
-        json={"status": "1", "message": "OK", "result": '{"abi": "some_abi"}'},
+        f"https://abidata.net/{KNOWN_ADDRESS}",
+        json=test_abi,
     )
     new_contract = scaffolder.from_block_explorer(KNOWN_ADDRESS, "new_contract")
     assert new_contract
-    assert new_contract.abi
+    assert new_contract.abi == test_abi
+    assert new_contract.address == KNOWN_ADDRESS
+    assert new_contract.name == "new_contract"
+    assert new_contract.author == "eightballer"
+
+    # Test network-specific scaffolding
+    responses.add(
+        responses.GET,
+        f"https://abidata.net/{KNOWN_NETWORK}/{KNOWN_ADDRESS}",
+        json=test_abi,
+    )
```

Comment:
Why 2 sets of networks calls overwritten.
Instead paramterise the methoid such that none network is given